tests/config.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"fmt"
+	"os"
+)
+
+var ConfigFile = ""
+var Config *KubeVirtTestsConfiguration
+
+func init() {
+	flag.StringVar(&ConfigFile, "config", "", "Path to a JSON formatted file from which the test suite will load its configuration. The path may be absolute or relative; relative paths start at the current working directory.")
+}
+
+// KubeVirtTestsConfiguration contains the configuration for KubeVirt tests
+type KubeVirtTestsConfiguration struct {
+	// StorageClass to use to create local PVCs
+	StorageClassLocal string `json:"StorageClassLocal"`
+	// StorageClass to use to create host-path PVCs
+	StorageClassHostPath string `json:"StorageClassHostPath"`
+	// StorageClass to use to create block-volume PVCs
+	StorageClassBlockVolume string `json:"StorageClassBlockVolume"`
+	// StorageClass to use to create rhel PVCs
+	StorageClassRhel string `json:"StorageClassRhel"`
+	// StorageClass to use to create windows PVCs
+	StorageClassWindows string `json:"StorageClassWindows"`
+}
+
+func NewKubeVirtTestsConfiguration() *KubeVirtTestsConfiguration {
+	return &KubeVirtTestsConfiguration{
+		StorageClassLocal:       "local",
+		StorageClassHostPath:    "host-path",
+		StorageClassBlockVolume: "block-volume",
+		StorageClassRhel:        "rhel",
+		StorageClassWindows:     "windows",
+	}
+}
+
+func loadConfig() *KubeVirtTestsConfiguration {
+	config := NewKubeVirtTestsConfiguration()
+
+	if ConfigFile != "" {
+		err := loadConfigFromFile(ConfigFile, config)
+
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Warning: error loading configuration file: %s\n", err)
```

Comment:
If we got up to here, that means user specified config file on command line explicitly. Does it make sense to continue when we are unable to read user's config file ?

Reply:
PR modified to panic instead of juste writing a message to stderr.